cmd/minishift/cmd/oc_env.go

```diff
@@ -57,6 +53,12 @@ func getOcShellConfig(api libmachine.API, ocPath string, forcedShell string, noP
 	}
 
 	cmdLine := "minishift oc-env"
+	if isProfileFlagUsed, profileName := getProfileNamefromFlag(); isProfileFlagUsed {
+		if profileName != profileActions.GetActiveProfile() {
+			cmdLine = fmt.Sprintf("minishift oc-env --profile=%s", profileName)
+		}
+	}
+
```

Comment:
@praveenkumar I do not think we need this change. `profileName` supposed to me give teh profile name that needs to be used.

Reply:
@praveenkumar I just tried below and it seems to work every time for me. 
```diff
diff --git a/cmd/minishift/cmd/oc_env.go b/cmd/minishift/cmd/oc_env.go
index fbe85a70..33a17302 100644
--- a/cmd/minishift/cmd/oc_env.go
+++ b/cmd/minishift/cmd/oc_env.go
@@ -31,7 +31,6 @@ import (
        "github.com/minishift/minishift/cmd/minishift/state"
        "github.com/minishift/minishift/pkg/minishift/cache"
        "github.com/minishift/minishift/pkg/minishift/config"
-       profileActions "github.com/minishift/minishift/pkg/minishift/profile"
        "github.com/minishift/minishift/pkg/util/os/atexit"
        "github.com/minishift/minishift/pkg/util/shell"
        "github.com/spf13/cobra"
@@ -104,14 +103,7 @@ var ocEnvCmd = &cobra.Command{
                util.ExitIfNotRunning(host.Driver, constants.MachineName)

                var shellCfg *OcShellConfig
-               var ocPath string
-               if constants.ProfileName != profileActions.GetActiveProfile() {
-                       // When PROFILE_NAME is not an active profile i.e. oc-env --profile PROFILE_NAME
-                       // is used we need to findout the oc path
-                       ocPath = getOcPath()
-               } else {
-                       ocPath = config.InstanceConfig.OcPath
-               }
+               var ocPath := config.InstanceConfig.OcPath
                shellCfg, err = getOcShellConfig(api, ocPath, forceShell, noProxy)
                if err != nil {
                        atexit.ExitWithMessage(1, fmt.Sprintf("Error running the oc-env command: %s", err.Error()))
```

Reply:
```
$ minishift oc-env
export PATH="/Users/lmohanty/.minishift/cache/oc/v3.7.0/darwin:$PATH"
# Run this command to configure your shell:
# eval $(minishift oc-env)

$ minishift profile list
- foo		Running
- minishift	Running		(Active)

$ minishift oc-env --profile foo
export PATH="/Users/lmohanty/.minishift/cache/oc/v3.9.0/darwin:$PATH"
# Run this command to configure your shell:
# eval $(minishift oc-env)
```

Reply:
@LalatenduMohanty I didn't get you the comment which @gbraad made is about having the `eval` output as per profile user provides. A user who is using the `foo` profile (active one) when try to execute `minishift oc-env` it suppose to provide `$(minishift oc-env)` but when `foo` is active and user try to get `oc-env` for a `bar` profile then it should show `$(minishift oc-env --profile bar)`.